redfish-core/lib/cable.hpp

```diff
@@ -3,11 +3,64 @@
 #include <query.hpp>
 #include <sdbusplus/asio/property.hpp>
 #include <sdbusplus/unpack_properties.hpp>
+#include <utils/chassis_utils.hpp>
 #include <utils/dbus_utils.hpp>
 #include <utils/json_utils.hpp>
 
 namespace redfish
 {
+
+/**
+ * @brief Api to get Cable Association.
+ * @param[in,out]   asyncResp       Async HTTP response.
+ * @param[in]       cableObjectPath Object path of the Cable with association.
+ * @param[in]       callback        callback method
+ */
+template <typename Callback>
+inline void
+    linkAssociatedCable(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& cableObjectPath, Callback&& callback)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, callback{std::forward<Callback>(callback)}](
+            const boost::system::error_code ec,
+            const std::variant<std::vector<std::string>>& endpoints) {
+        if (ec)
+        {
+            if (ec.value() == EBADR)
+            {
+                // This cable have no association.
+                BMCWEB_LOG_DEBUG << "No association found";
+                return;
+            }
+            BMCWEB_LOG_ERROR << "DBUS response error" << ec.message();
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        const std::vector<std::string>* value =
+            std::get_if<std::vector<std::string>>(&(endpoints));
+
+        if (value == nullptr)
+        {
+            BMCWEB_LOG_DEBUG << "Error getting cable association!";
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        if ((*value).empty())
+        {
+            BMCWEB_LOG_DEBUG << "No association found for cable";
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        callback(*value);
+        },
+        "xyz.openbmc_project.ObjectMapper", cableObjectPath,
+        "org.freedesktop.DBus.Properties", "Get",
```

Comment:
Upstream we are using 
        sdbusplus::asio::getProperty
There is also a utility function you could pull in at https://gerrit.openbmc.org/c/openbmc/bmcweb/+/60872 
Up to you if you want to move

Reply:
I've changed to use dbus::utility::getAssociationEndPoints().